clients/src/test/java/org/apache/kafka/clients/consumer/ConsumerPartitionAssignorTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.consumer;
+
+
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+
+import static org.apache.kafka.clients.consumer.ConsumerPartitionAssignor.getAssignorInstances;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ConsumerPartitionAssignorTest {
+
+    private List<String> classNames;
+    private List<Object> classTypes;
+
+    @Test
+    public void shouldInstantiateAssignor() {
+        classNames = Collections.singletonList(StickyAssignor.class.getName());
+        List<ConsumerPartitionAssignor> assignors = getAssignorInstances(classNames, Collections.emptyMap());
+        assertTrue(assignors.get(0) instanceof StickyAssignor);
+    }
+
+    @Test
+    public void shouldInstantiateListOfAssignors() {
+        classNames = Arrays.asList(StickyAssignor.class.getName(), CooperativeStickyAssignor.class.getName());
+        List<ConsumerPartitionAssignor> assignors = getAssignorInstances(classNames, Collections.emptyMap());
+        assertTrue(assignors.get(0) instanceof StickyAssignor);
+    }
+
+    @Test
+    public void shouldThrowKafkaExceptionOnNonAssignor() {
+        classNames = Collections.singletonList(String.class.getName());
+        assertThrows(KafkaException.class, () -> getAssignorInstances(classNames, Collections.emptyMap()));
+    }
+
+    @Test
+    public void shouldThrowKafkaExceptionOnAssignorNotFound() {
+        classNames = Collections.singletonList("Non-existent assignor");
+        assertThrows(KafkaException.class, () -> getAssignorInstances(classNames, Collections.emptyMap()));
+    }
+
+    @Test
+    public void shouldInstantiateFromClassType() {
```

Comment:
I think it would make sense to style this test (and `shouldInstantiateFromListOfClassTypes` below) more like `shouldInstantiateAssignors` now, ie where we actually validate the assignors that are returned (eg `assertTrue(assignors.get(0) instanceof StickyAssignor)`). Previously this test was just making sure that we adaptor would work and we wouldn't throw an exception when constructing the consumer, that's why it's like this

Reply:
`assignors ` is not visible in `KafkaConsumer`, we should add a `getAssignors()` in `KafkaConsumer` to test this. Maybe the other tests are enough to verify that the `getAssignorInstances` operate rightly, How about remove these 2 test cases?

Reply:
Ah, I was suggesting to just replicate the `shouldInstantiateAssignor` and `shouldInstantiateListOfAssignors` tests exactly, but with the `classTypes` being eg `StickyAssignor.class` instead of `StickyAssignor.class.getName()`. For example

```
classNames = Collections.singletonList(StickyAssignor.class);
        List<ConsumerPartitionAssignor> assignors = getAssignorInstances(classNames, Collections.emptyMap());
        assertTrue(assignors.get(0) instanceof StickyAssignor);
```

Reply:
It seems that `getAssignorInstances()` can only accept `List<String>` as the first parameter, so this could not work.😳

Reply:
Ah, yeah, you'd need to do something more like what actually happens in the KafkaConsumer/`getAssignorInstances` code. eg
```
@Test
@SuppressWarnings("unchecked")
public void shouldInstantiateAssignorClass() {
    Object classTypes = Collections.singletonList(StickyAssignor.class);
    List<ConsumerPartitionAssignor> assignors = getAssignorInstances((List<String>) classTypes, Collections.emptyMap());
     assertTrue(assignors.get(0) instanceof StickyAssignor);
}
```

Reply:
It's super awkward, obviously, but since this is what happens when we process the configs in the real code we should try to replicate that in the test

Reply:
Yes, this awkward, I fixed this with some code refactoring and fixed the comment in upgrade.html, PTAL, Thank you for your patience.

Reply:
Nice, thanks for the update. Looks good